lxd/network/acl/acl_ovn.go

```diff
@@ -86,12 +86,43 @@ func OVNEnsureACLs(s *state.State, logger logger.Logger, client *openvswitch.OVN
 
 	var err error
 	var projectID int64
+	peerTargetNetIDs := make(map[string]int64)
 	err = s.Cluster.Transaction(func(tx *db.ClusterTx) error {
 		projectID, err = tx.GetProjectID(aclProjectName)
+		if err != nil {
+			return fmt.Errorf("Failed getting project ID for project %q: %w", aclProjectName, err)
+		}
+
+		// Build a mapping of "<network_name>/<peer_name>" names to target network IDs.
+		q := `SELECT p.name, n.name, p.target_network_id
+			FROM networks_peers AS p
+			JOIN networks AS n ON n.id = p.network_id
+			WHERE n.project_id = ?
+			AND n.type = ?
+			AND p.target_network_id > 0
+		`
+		err = tx.QueryScan(q, func(scan func(dest ...interface{}) error) error {
```

Comment:
Running a DB query from outside of `lxd/db` feels wrong. Any reason why this isn't a new function in the db package?

Reply:
I considered different approaches. I went for this approach as I needed an ovn specific format for the return value. Otherwise id need to add a specific struct return type. Which I'm fine with if preferred.

Reply:
I'd definitely prefer we keep all SQL to the db package (which is mostly the case currently, patches.go is the exception).

Maybe I'm missing something while reading this code, but it looks like we could have a:
```
GetPeerTargetNetworkIDs(projectName string, networkType int) map[string]int64
```

In the DB package and just call that from here?

Reply:
Yes that would work but didn't want to encode the network and peer name into the map string component, so would use a new struct type.

Reply:
I've refactored this now as discussed.